### PR TITLE
[Feature] Add option to select one row at a time in data tables

### LINKED ISSUE #2960: [Feature Request] Add option to only select one row at a time in data tables
### New Functionality
Option to select one row at a time in a data table, radio-button style. Selecting a new row will clear the previous selection. 

### Improvements
No existing workaround

### Bugs or Edge Cases it Helps Avoid
Allows the user to provide extra functionality in custom components based on a guaranteed single-row selection in the data table; a preview pane, for example.

### PR DESCRIPTION
**Pull request by jim0thy**
## Description

Option to select one row at a time in a data table, radio-button style. Selecting a new row will clear the previous selection.
Adds `radioSelect` boolean prop to `v-data-table`, plus code to clear previous selection if the prop is true.

## Motivation and Context
Allows the user to provide extra functionality in custom components based on a guaranteed single-row selection in the data table; a preview pane, for example.

Resolves #2960 

## How Has This Been Tested?
Jest. 
No new test, as modified code on`data-iterable` isn't currently subject to unit testing

## Markup:
````html
<template>
  <v-data-table
    v-bind:headers="headers"
    v-bind:items="items"
    v-bind:search="search"
    v-model="selected"
    item-key="name"
    radio-select
    class="elevation-1"
  >
    <template slot="headerCell" slot-scope="props">
      <v-tooltip bottom>
        <span slot="activator">
          {{ props.header.text }}
        </span>
        <span>
          {{ props.header.text }}
        </span>
      </v-tooltip>
    </template>
    <template slot="items" slot-scope="props">
      <td>
        <v-checkbox
          primary
          hide-details
          v-model="props.selected"
        ></v-checkbox>
      </td>
      <td>{{ props.item.name }}</td>
      <td class="text-xs-right">{{ props.item.calories }}</td>
      <td class="text-xs-right">{{ props.item.fat }}</td>
      <td class="text-xs-right">{{ props.item.carbs }}</td>
      <td class="text-xs-right">{{ props.item.protein }}</td>
      <td class="text-xs-right">{{ props.item.sodium }}</td>
      <td class="text-xs-right">{{ props.item.calcium }}</td>
      <td class="text-xs-right">{{ props.item.iron }}</td>
    </template>
  </v-data-table>
</template>

<script>
  export default {
    data () {
      return {
        search: '',
        selected: [],
        headers: [
          {
            text: 'Dessert (100g serving)',
            align: 'left',
            sortable: false,
            value: 'name'
          },
          { text: 'Calories', value: 'calories' },
          { text: 'Fat (g)', value: 'fat' },
          { text: 'Carbs (g)', value: 'carbs' },
          { text: 'Protein (g)', value: 'protein' },
          { text: 'Sodium (mg)', value: 'sodium' },
          { text: 'Calcium (%)', value: 'calcium' },
          { text: 'Iron (%)', value: 'iron' }
        ],
        items: [...]
      }
    }
  }
</script>
````

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: src/mixins/data-iterable.js [hunk 1]
```diff
@@ -275,7 +276,7 @@ export default {
       )
 
       return this.hideActions &&
-        !this.hasPagination
+      !this.hasPagination
```
Comment (nekosaur) [inline src/mixins/data-iterable.js]: fix this

File: src/mixins/data-iterable.js [hunk 2]
```diff
@@ -316,8 +317,15 @@ export default {
           }
 
           let selected = this.value.slice()
-          if (value) selected.push(item)
-          else selected = selected.filter(i => i[keyProp] !== itemKey)
+          if (value) {
+            if (this.radioSelect) {
+              selected.length = 0
```
Comment (nekosaur) [inline src/mixins/data-iterable.js]: can be simplified to `selected = [item]`

**Comment by Dylan-Chapman** [comment]
Faced this issue recently. Got around it by having ```selected``` be a number instead of an array, and having ```<v-checkbox :input-value="value" :value="props.item.id"></v-checkbox>```. Would love to have a built-in solution to it.

**Comment by jim0thy** [comment]
@nekosaur `single-select` makes more sense given the context. I'll update the prop name.

I'm working on some tests at the moment; trying to decide whether to run them against the `v-data-table` component or the `data-iterable` mixin.

**Comment by jim0thy** [comment]
@neckosaur I'm not sure this can be tested properly using the current version of `vue-test-utils`. A full test requires the `v-data-table` component to me mounted with a scoped slot, which `vue-test-utils` can't currently handle.

There's an open issue here discussing the problem: vuejs/vue-test-utils#156

**Comment by KaelWD** [comment]
We're still using avoriaz, but that probably has the same issue

**Comment by johnleider** [comment]
Closing due to inactivity.

**Pull request closed without merging**